### DISCUSSION TITLE
Suggestion for adding a plugin similar to hexo-abbrlink

### DISCUSSION THREAD

**Discussion by yinyu985** [vuepress-theme-hope/vuepress-theme-hope#3044] [Ideas]
I am using vuepress-hope theme for my website and I really like it. Thanks for creating such a wonderful theme! I was wondering if you would be interested in adding a plugin similar to hexo-abbrlink, which generates short random links for each post. I have tried some existing plugins, but they are not well maintained or have a poor user experience. It would be great to have a high-quality plugin integrated with the vuepress-hope theme.
Let me know if you're interested in this suggestion. Thank you for your hard work on this amazing theme!

**Comment by Mister-Hope** [comment]
I think you can set a permalink for each page based on their path in `extendsPage` lifecycle, that's pretty easy and only cost a few lines, contact me with further details if you have any trouble doing that.
